### PR TITLE
[4258][FIX] template_content_swapper

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
[4258](https://www.quartile.co/web#id=4258&cids=3&menu_id=505&action=1457&model=project.task&view_type=form)

### CODE REVIEW

File: template_content_swapper/models/ir_ui_view.py [hunk 1]
```diff
@@ -16,7 +16,8 @@ def _render_template(self, template, values=None):
         result_str = str(result)
         lang_code = "en_US"
         request = values.get("request")
```
Comment (kanda999) [inline template_content_swapper/models/ir_ui_view.py]:
```suggestion
```

**Review by kanda999 — CHANGES_REQUESTED** [review]

**Review by yostashiro — COMMENTED** [review]
The scope of `_render_template()` is not clear to me when mega menu is involved.  Is it processed on the mega menu only?  If that is the case, is it still relevant for content swapping at all?

**Comment by AungKoKoLin1997** [comment]
> The scope of `_render_template()` is not clear to me when mega menu is involved. Is it processed on the mega menu only? If that is the case, is it still relevant for content swapping at all?

@yostashiro Our scope for content swapping is for all views and reports. So, _render_template() is called every time the view is rendered.

**Comment by AungKoKoLin1997** [comment]
I checked the log when I created the mega menu. It seems like the _render_templated is called twice and first one is in cache and there is no values. The first view is `s_mega_menu_odoo_menu` and second view is home page view id.
```
2024-08-13 05:48:15,545 599764 INFO 5stx3 root: ir.ui.view(1351,) 
2024-08-13 05:48:15,545 599764 INFO 5stx3 root: None 
2024-08-13 05:48:15,550 599764 INFO 5stx3 odoo.modules.registry: At least one model cache has been invalidated, signaling through the database. 
2024-08-13 05:48:15,554 599764 INFO 5stx3 werkzeug: 58.11.156.73 - - [13/Aug/2024 05:48:15] "POST /web/dataset/call_kw/website.menu/save HTTP/1.0" 200 - 95 0.031 0.111
2024-08-13 05:48:16,896 599764 INFO 5stx3 root: ir.ui.view(1141,) 
2024-08-13 05:48:16,896 599764 INFO 5stx3 root: {'deletable': True, 'main_object': website.page(4,), 'response_template': 1141, 'request': <odoo.http.HttpRequest object at 0x7f26ba76a070>, 'backend_menu_id': 265}
```

**Comment by yostashiro** [comment]
@AungKoKoLin1997 My question was more from the business perspective.  Mega menu is totally customizable IIUC, and if that is the case, would there be any point doing the content swapping at all?  If there was no point, I thought one way of handling this case was to just return result if there there was no value.  Not saying that this should be the conclusion.

**Comment by AungKoKoLin1997** [comment]
> If there was no point, I thought one way of handling this case was to just return result if there there was no value.

@yostashiro I got the point. When I quick check for other options in website and I see mega menu is only the case. So, I will take your suggestion.

File: template_content_swapper/models/ir_ui_view.py [hunk 2]
```diff
@@ -18,11 +18,10 @@ def _render_template(self, template, values=None):
...
             # For views
             lang_code = values.get("request").env.lang
-        else:
-            # For reports or views when there is no request environment
+        elif re.search(r'data-oe-lang="([^"]+)"', result_str):
+            # For reports
             lang_match = re.search(r'data-oe-lang="([^"]+)"', result_str)
-            if lang_match:
-                lang_code = lang_match.group(1)
+            lang_code = lang_match.group(1)
```
Comment (yostashiro) [inline template_content_swapper/models/ir_ui_view.py]:
```.py
        lang_match = re.search(r'data-oe-lang="([^"]+)"', result_str)
        if lang_match:
            # For reports
            lang_code = lang_match.group(1)
```

File: template_content_swapper/models/ir_ui_view.py [hunk 3]
```diff
@@ -14,16 +14,14 @@ class IrUiView(models.Model):
...
-            lang_code = request.env.lang
-        else:
+            lang_code = values.get("request").env.lang
+        lang_match = re.search(r'data-oe-lang="([^"]+)"', result_str)
+        if lang_match:
             # For reports
-            lang_match = re.search(r'data-oe-lang="([^"]+)"', result_str)
-            if lang_match:
-                lang_code = lang_match.group(1)
+            lang_code = lang_match.group(1)
```
Comment (yostashiro) [inline template_content_swapper/models/ir_ui_view.py]: Sorry to correct many times...
```.py
        else:
            lang_match = re.search(r'data-oe-lang="([^"]+)"', result_str)
            if lang_match:
                # For reports
                lang_code = lang_match.group(1)
```

**Review by yostashiro — APPROVED** [review]
Code review.

**Review by kanda999 — APPROVED** [review]
Functioanl and Code review.

Could you please update the OCA PR.

**Pull request merged**